crates/solvers-dto/src/auction.rs

```diff
@@ -33,11 +33,22 @@ pub struct Order {
     #[serde_as(as = "HexOrDecimalU256")]
     pub sell_amount: U256,
     #[serde_as(as = "HexOrDecimalU256")]
+    pub full_sell_amount: U256,
+    #[serde_as(as = "HexOrDecimalU256")]
     pub buy_amount: U256,
+    #[serde_as(as = "HexOrDecimalU256")]
+    pub full_buy_amount: U256,
+    pub fee_policies: Option<Vec<FeePolicy>>,
+    pub valid_to: u32,
     pub kind: Kind,
+    pub receiver: Option<H160>,
+    pub owner: H160,
     pub partially_fillable: bool,
+    pub pre_interactions: Vec<model::interaction::InteractionData>,
+    pub post_interactions: Vec<model::interaction::InteractionData>,
     pub class: Class,
-    pub fee_policies: Option<Vec<FeePolicy>>,
+    #[serde(flatten)]
+    pub signature: model::signature::Signature,
```

Comment:
Since we only need to serialize the data here I think it's fine to not import the `model` crate here and instead copy the few signature related types.

Reply:
@MartinquaXD it is not "few" types. It is all the deserialization, with `SignatureSchema` and everything. I did copy it and it was horrible. I am totally against it.

Reply:
> It is all the deserialization

But we don't really have to deserialize the signatures into anything fancy.
To me it seems like we actually just need:
```rust
#[derive(Serialize, Deserialize)]
enum SigningScheme {
    Eip712,
    EthSign,
    Eip1271,
    PreSign,
}
```

And the struct we send over the wire would be:
```
```

Reply:
ah, so you mean to do it like that, ok ok. Won't it be needed in the solvers the whole `Signature` functionality in the future? :thinking:

Reply:
So far none of the solvers we have to maintain are using the signature at all (or any other field you are adding in this PR). It's exclusively external solvers that use those and they have to do their own deserialization anyway. Also I think even for their purposes this simple version should be good enough given that this is how they currently "consume" the signature from the `/auction` endpoint.